xcm/pallet-xcm-benchmarks/src/generic/benchmarking.rs

```diff
@@ -0,0 +1,311 @@
+		let max_response_weight = Default::default();
+
+		let instruction = Instruction::ReportError { query_id, dest, max_response_weight };
+		let xcm = Xcm(vec![instruction]);
+	}: {
+		executor.execute(xcm)?;
+	} verify {
+		// the execution succeeding is all we need to verify this xcm was successful
+	}
+
+	claim_asset {
+		let (origin, ticket, assets) = T::claimable_asset()
+			.ok_or(BenchmarkError::Skip)?;
+
+		// We place some items into the asset trap to claim.
+		let mut executor = new_executor::<T>(origin.clone());
+		executor.holding = assets.clone().into();
+		executor.execute(Xcm(vec![]))?;
+		executor.post_execute(0);
+		// Assets should be in the trap now.
+
+		let mut executor = new_executor::<T>(origin);
+		let instruction = Instruction::ClaimAsset { assets: assets.clone(), ticket };
+		let xcm = Xcm(vec![instruction]);
+	} :{
+		executor.execute(xcm)?;
+	} verify {
+		assert!(executor.holding.ensure_contains(&assets).is_ok());
+	}
+
+	trap {
+		let mut executor = new_executor::<T>(Default::default());
+		let instruction = Instruction::Trap(10);
+		let xcm = Xcm(vec![instruction]);
+	} : {
+		match executor.execute(xcm) {
+			Err(error) if error.xcm_error == XcmError::Trap(10) => {
+				// This is the success condition
+			},
+			_ => return Err(BenchmarkError::Stop("xcm trap did not return the expected error"))
+		};
+	} verify {
+		// Verification is done above.
+	}
+
+	subscribe_version {
+		use xcm_executor::traits::VersionChangeNotifier;
+		let origin = T::transact_origin().ok_or(BenchmarkError::Skip)?;
+		let query_id = Default::default();
+		let max_response_weight = Default::default();
+		let mut executor = new_executor::<T>(origin.clone());
+		let instruction = Instruction::SubscribeVersion { query_id, max_response_weight };
+		let xcm = Xcm(vec![instruction]);
+	} : {
+		executor.execute(xcm)?;
+	} verify {
+		assert!(<T::XcmConfig as xcm_executor::Config>::SubscriptionService::is_subscribed(&origin));
+	}
+
+	unsubscribe_version {
+		use xcm_executor::traits::VersionChangeNotifier;
+		// First we need to subscribe to notifications.
+		let origin = T::transact_origin().ok_or(BenchmarkError::Skip)?;
+		let query_id = Default::default();
+		let max_response_weight = Default::default();
+		let mut executor = new_executor::<T>(origin.clone());
+		let instruction = Instruction::SubscribeVersion { query_id, max_response_weight };
+		let xcm = Xcm(vec![instruction]);
+		executor.execute(xcm)?;
+		assert!(<T::XcmConfig as xcm_executor::Config>::SubscriptionService::is_subscribed(&origin));
+
+		let mut executor = new_executor::<T>(origin.clone());
+		let instruction = Instruction::UnsubscribeVersion;
+		let xcm = Xcm(vec![instruction]);
+	} : {
+		executor.execute(xcm)?;
+	} verify {
+		assert!(!<T::XcmConfig as xcm_executor::Config>::SubscriptionService::is_subscribed(&origin));
+	}
+
+	initiate_reserve_withdraw {
+		let assets = MultiAssetFilter::Wild(All);
```

Comment:
Same issue with `BuyExecution`, it calls `Assets::general_take`, which doesn't necessarily have the worst case if you have an `All` wildcard.

Reply:
This one's a bit more annoying to track down, because it also depends on the kind of assets that already exist in holding, and the executor has to [reanchor these assets](https://github.com/paritytech/polkadot/blob/c198d26d3fdb4b3e26599059335a711c4bc9e3b1/xcm/xcm-executor/src/lib.rs#L410), i.e. _where_ the assets originated matters in the worst case as well.

Reply:
added worst case holding and added a multiassetfilter where we look for every asset in holding explicitly.